### DISCUSSION TITLE
getTransactionInfo Returning encrypted payload

### DISCUSSION THREAD

**Discussion by jadayash** [imdhemy/laravel-in-app-purchases#181] [Q&A] [answered]
Hello
The getTransactionInfo is returning encrypted data, how can i decrypt this please?

`        $subscription = $event->getServerNotification()->getSubscription();
        $data = $subscription->getProviderRepresentation();
        $transactionInfo = $data->getTransactionInfo();
`

This what is being returned:

`{"\u0000Imdhemy\\AppStore\\ValueObjects\\JwsTransactionInfo\u0000jws":{"Imdhemy\\AppStore\\Jws\\Jws":"eyJhbGciOiJFUzI1NiIsIng1YyI6WyJNSUlFTURDQ0E3YWdBd0lCQWdJUWFQb1BsZ`

**Comment by imdhemy** [comment, marked as answer]
Hello 

You don't have to decrypt them, just use the getter methods:

```php
$transactionInfo = $data->getTransactionInfo();

// To get data as an array
$claims = $transactionInfo->getClaims();

// still you can access data by name using a getter method
$envitonment = $transactionInfo->getEnvironment();
$expiresDate = $transactionInfo->getExpiresDate();

// and so on ...
```
